File: ebs_flutter/lib/model/Employee.dart
class Employee {
  final int? id;
  final int? employeeCode;
  final String? firstName;
  final String? lastName;
  final String? gender;
  final String? phone;
  final String? email;
  final String? address;
  final DateTime? dateOfBirth;
  final DateTime? hireDate;
  final double? salary;
  final int? departmentCode;
  final String? departmentName;
  final int? designationCode;
  final String? designationName;
  final String? status;

  Employee({
    this.id,
    this.employeeCode,
    this.firstName,
    this.lastName,
    this.gender,
    this.phone,
    this.email,
    this.address,
    this.dateOfBirth,
    this.hireDate,
    this.salary,
    this.departmentCode,
    this.departmentName,
    this.designationCode,
    this.designationName,
    this.status,
  });

  factory Employee.fromJson(Map<String, dynamic> json) {
    return Employee(
      id: json['id'],
      employeeCode: json['employeeCode'],
      firstName: json['firstName'],
      lastName: json['lastName'],
      gender: json['gender'],
      phone: json['phone'],
      email: json['email'],
      address: json['address'],
      dateOfBirth: json['dateOfBirth'] != null ? DateTime.parse(json['dateOfBirth']) : null,
      hireDate: json['hireDate'] != null ? DateTime.parse(json['hireDate']) : null,
      salary: json['salary'] != null ? (json['salary'] as num).toDouble() : null,
      departmentCode: json['departmentCode'],
      departmentName: json['departmentName'],
      designationCode: json['designationCode'],
      designationName: json['designationName'],
      status: json['status'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'employeeCode': employeeCode,
      'firstName': firstName,
      'lastName': lastName,
      'gender': gender,
      'phone': phone,
      'email': email,
      'address': address,
      'dateOfBirth': dateOfBirth?.toIso8601String(),
      'hireDate': hireDate?.toIso8601String(),
      'salary': salary,
      'departmentCode': departmentCode,
      'departmentName': departmentName,
      'designationCode': designationCode,
      'designationName': designationName,
      'status': status,
    };
  }
}

File: ebs-springboot/src/main/java/com/erp/model/Employee.java
package com.erp.model;

import javax.persistence.*;
import java.util.Date;

@Entity(name = "employee")
@Table(name = "employee")
public class Employee {

	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private Long id;

	@Column(name = "employee_code")
	private Long employeeCode;

	@Column(name = "first_name")
	private String firstName;

	@Column(name = "last_name")
	private String lastName;

	@Column(name = "gender")
	private String gender;

	@Column(name = "phone")
	private String phone;

	@Column(name = "email")
	private String email;

	@Column(name = "address")
	private String address;

	@Column(name = "date_of_birth")
	@Temporal(TemporalType.DATE)
	private Date dateOfBirth;

	@Column(name = "hire_date")
	@Temporal(TemporalType.DATE)
	private Date hireDate;

	@Column(name = "salary")
	private Double salary;

	@Column(name = "department_code")
	private Long departmentCode;

	@Column(name = "department_name")
	private String departmentName;

	@Column(name = "designation_code")
	private Long designationCode;

	@Column(name = "designation_name")
	private String designationName;

	@Column(name = "status")
	private String status;

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public Long getEmployeeCode() {
		return employeeCode;
	}

	public void setEmployeeCode(Long employeeCode) {
		this.employeeCode = employeeCode;
	}

	public String getFirstName() {
		return firstName;
	}

	public void setFirstName(String firstName) {
		this.firstName = firstName;
	}

	public String getLastName() {
		return lastName;
	}

	public void setLastName(String lastName) {
		this.lastName = lastName;
	}

	public String getGender() {
		return gender;
	}

	public void setGender(String gender) {
		this.gender = gender;
	}

	public String getPhone() {
		return phone;
	}

	public void setPhone(String phone) {
		this.phone = phone;
	}

	public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

	public String getAddress() {
		return address;
	}

	public void setAddress(String address) {
		this.address = address;
	}

	public Date getDateOfBirth() {
		return dateOfBirth;
	}

	public void setDateOfBirth(Date dateOfBirth) {
		this.dateOfBirth = dateOfBirth;
	}

	public Date getHireDate() {
		return hireDate;
	}

	public void setHireDate(Date hireDate) {
		this.hireDate = hireDate;
	}

	public Double getSalary() {
		return salary;
	}

	public void setSalary(Double salary) {
		this.salary = salary;
	}

	public Long getDepartmentCode() {
		return departmentCode;
	}

	public void setDepartmentCode(Long departmentCode) {
		this.departmentCode = departmentCode;
	}

	public String getDepartmentName() {
		return departmentName;
	}

	public void setDepartmentName(String departmentName) {
		this.departmentName = departmentName;
	}

	public Long getDesignationCode() {
		return designationCode;
	}

	public void setDesignationCode(Long designationCode) {
		this.designationCode = designationCode;
	}

	public String getDesignationName() {
		return designationName;
	}

	public void setDesignationName(String designationName) {
		this.designationName = designationName;
	}

	public String getStatus() {
		return status;
	}

	public void setStatus(String status) {
		this.status = status;
	}

}
